tenacity/__init__.py

```diff
@@ -279,6 +279,14 @@ def statistics(self) -> t.Dict[str, t.Any]:
             self._local.statistics = t.cast(t.Dict[str, t.Any], {})
             return self._local.statistics
 
+    @property
+    def iter_state(self) -> t.Dict[str, t.Any]:
+        try:
+            return self._local.iter_state  # type: ignore[no-any-return]
+        except AttributeError:
+            self._local.iter_state = t.cast(t.Dict[str, t.Any], {})
```

Comment:
Can we have something either more structured or split?
I am no fan of having a dict with `Any` and random keys TBH.

At least a `TypedDict`?

Reply:
Sounds fair, wdyt? https://github.com/jd/tenacity/pull/434/commits/538686b75bd90f15c6c7ead7999904620fa1f910

tbh I just copied the approach from the `statistics` property, and I was tempted to change the `local()` implementation for a `ContextVar` and maybe introducing `dataclass`es, but seemed like a bit too much to do together with these changes. Maybe something to tackle separately if that's something you'd like to see.